collector/lib/CollectorConfig.cpp

```diff
@@ -185,6 +188,21 @@ void CollectorConfig::InitCollectorConfig(CollectorArgs* args) {
                   }
                 });
 
+  std::for_each(detailed_networks.value().begin(), detailed_networks.value().end(),
+                [&detailed_networks = this->detailed_networks_](const std::string& str) {
+                  if (str.empty())
+                    return;
+
+                  std::optional<IPNet> net = IPNet::parse(str);
+
+                  if (net) {
+                    CLOG(INFO) << "Detail network : " << *net;
+                    detailed_networks.emplace_back(std::move(*net));
+                  } else {
+                    CLOG(ERROR) << "Invalid network in ROX_DETAIL_NETWORKS : " << str;
+                  }
+                });
```

Comment:
I have updated this part to make use of simple for(:) loops. Since there was an other very similar loop for ignored_networks, I changed this one as well.

Reply:
I didn't know there was another block exactly like this one, now I kinda want to put the for loop in a lambda and call it twice passing the input vector and the output member, but this is fine as well.